api/workspace.go

```diff
@@ -0,0 +1,175 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	cloud "github.com/mattermost/mattermost-cloud/model"
+)
+
+// initWorkspace registers workspace endpoints on the given router.
+func initWorkspace(apiRouter *mux.Router, context *Context) {
+	workspacesRouter := apiRouter.PathPrefix("/workspaces").Subrouter()
+	workspacesRouter.Handle("/list", newAPIHandler(context, handleListWorkspaces)).Methods("POST")
```

Comment:
Is it common to use POST to do a list of resources? Feels a bit odd to me, but 0/5

Reply:
When there's a ton of parameters I prefer to include them in the request body rather than the URL. It's similar to how we do our search endpoints in the MM server. I'm maybe 1/5 but can be swayed either way